src/DssVest.sol

```diff
@@ -223,6 +238,17 @@ abstract contract DssVest is ERC2771Context, Initializable {
     function claim(bytes32 _bch, address _usr, uint256 _tot, uint256 _bgn, uint256 _tau, uint256 _eta, address _mgr, bytes32 _slt) public lock returns (uint256 id) {
         require(_bch == keccak256(abi.encodePacked(_usr, _tot, _bgn, _tau, _eta, _mgr, _slt)), "DssVest/invalid-hash");
         require(commitments[_bch], "DssVest/commitment-not-found");
+        uint48 revocationTime = toUint48(revocations[_bch]);
+        if ( revocationTime < _bgn + _eta  ) {
+            // commitment has been revoked before the cliff: vesting plan is cancelled
+            require(revocationTime == 0, "DssVest/commitment-revoked-before-cliff");
+        } else if ( revocationTime < _bgn + _tau ) {
+            // commitment has been revoked after the cliff, but before the end: vesting plan values have to be updated
+            // goal: behave as if the vesting plan was created when committed, and yanked when revoked
+            _tot = mul(_tot, sub(revocationTime, _bgn)) / _tau; // newTot as amount accrued if yanked at revocationTime
```

Comment:
did you double-check the math when it comes to rounding errors?

Reply:
I copied the math from the `accrued` function:
```
amt = mul(_tot, sub(_time, _bgn)) / sub(_fin, _bgn);
```
`_time` is `revocationTime` in our case, and `sub(_fin, _bgn)` is `_tau`.

We could use accrued instead, [like in this PR](https://github.com/corpus-io/dss-vest/pull/52), but that would do some unnecessary operations.